### PR TITLE
wrong_self_convention: `to_` convention respects `Copy` types

### LINKED ISSUE #6727: wrong_self_convention possibly too strict for Copy types
Lint name: wrong_self_convention

I tried this code:

```rust
trait F32Ext {
    fn to_something_else(self) -> f32;
}
```

I get this:
```
warning: methods called `to_*` usually take self by reference; consider choosing a less ambiguous name
  --> src/main.rs:66:21
   |
66 |     fn to_some_unit(self) -> f32;
   |                     ^^^^
   |
```

If this trait is implemented for types such as `f32`, then i think the `to_` prefix makes complete sense - `f32` already has methods with that prefix and it makes no sense to take such a small type by reference.

I think this lint should allow `to_` on Copy types to take `self` without reference.

### Meta

`cargo clippy -V`: clippy 0.1.52 (e9920ef 2021-02-11)
`rustc -Vv`:
```
rustc 1.52.0-nightly (e9920ef77 2021-02-11)
binary: rustc
commit-hash: e9920ef7749d11fc71cc32ca4ba055bcfeaab945
commit-date: 2021-02-11
host: x86_64-unknown-linux-gnu
release: 1.52.0-nightly
LLVM version: 11.0.1
```

### PR DESCRIPTION
**Pull request by mgacek8**
fixes #6727 
changelog: wrong_self_convention: `to_` convention respects `Copy` types

### CODE REVIEW

File: clippy_lints/src/methods/mod.rs [hunk 1]
```diff
@@ -193,14 +193,18 @@ declare_clippy_lint! {
...
-    /// |-------|------------|----------------------|
-    /// |`as_`  | none       |`&self` or `&mut self`|
-    /// |`from_`| none       | none                 |
-    /// |`into_`| none       |`self`                |
-    /// |`is_`  | none       |`&self` or none       |
-    /// |`to_`  | `_mut`     |`&mut &self`          |
-    /// |`to_`  | not `_mut` |`&self`               |
+    /// |Prefix |Postfix     |`self` taken           | `self` type  |
+    /// |-------|------------|-----------------------|--------------|
+    /// |`as_`  | none       |`&self` or `&mut self` | any          |
```
Comment (mgacek8) [inline clippy_lints/src/methods/mod.rs:198]: I'm wondering if it wouldn't be more clear to have `n/a` instead of `none` where the Postfix doesn't matter.

Comment (llogiq) [reply clippy_lints/src/methods/mod.rs:198]: I personally don't think an abbreviation of "not applicable" is clearer than "none" when there is no suffix, but that may be my dislike of needless abbreviations.

File: tests/ui/use_self.fixed [hunk 2]
```diff
@@ -79,6 +79,7 @@ mod issue2894 {
     }
 
     // This should not be linted
```
Comment (llogiq) [inline tests/ui/use_self.fixed:81]: If this comment is correct, then why the `#[allow(..)]` below?

File: tests/ui/use_self.rs [hunk 3]
```diff
@@ -79,6 +79,7 @@ mod issue2894 {
     }
 
     // This should not be linted
```
Comment (llogiq) [inline tests/ui/use_self.rs:81]: Same here. I get that it's probably a different lint that should not trigger, but it still feels wrong.

**Review by llogiq — COMMENTED** [review]
This looks pretty solid, I just have one nitpick in the tests.

File: tests/ui/use_self.rs (hunk 3 above)
Comment (mgacek8) [reply tests/ui/use_self.rs:81]: Yes, the comment says that `use_self` shouldn't trigger. 
I added the `#[allow(clippy::wrong_self_convention)]` to suppress the warning that occurred there, after current changes. I wasn't sure whether I should adjust the self convention there.
Agree it's better to adjust that uitest to the proper self convention, which I did in the next commit.

File: tests/ui/wrong_self_convention.stderr [hunk 4]
```diff
@@ -191,5 +175,21 @@ LL |         fn from_i32(self);
    |
    = help: consider choosing a less ambiguous name
 
+error: methods with the following characteristics: (`to_*` and `self type is  Copy`) usually take self by value
```
Comment (llogiq) [inline tests/ui/wrong_self_convention.stderr:178]: Looking once more at the resulting error message, this gave me pause: "\`self type is  Copy\`" should probably read "\`self\` type is \`Copy\`" (note the double space and the backticks).

**Review by llogiq — COMMENTED** [review]
I had another look and found another nit to pick 😉

File: tests/ui/wrong_self_convention.stderr (hunk 4 above)
Comment (mgacek8) [reply tests/ui/wrong_self_convention.stderr]: Good catch, thanks!

File: tests/ui/wrong_self_convention.stderr [hunk 5]
```diff
@@ -7,189 +7,189 @@ LL |     fn from_i32(self) {}
...
+error: methods called "is_*" usually take self by reference or no self
   --> $DIR/wrong_self_convention.rs:40:15
    |
 LL |     fn is_i32(self) {}
    |               ^^^^
    |
    = help: consider choosing a less ambiguous name
 
-error: methods called `to_*` usually take self by reference
+error: methods with the following characteristics: ("to_*" and "`self` type is not `Copy`") usually take self by reference
```
Comment (llogiq) [inline tests/ui/wrong_self_convention.stderr:42]: Oops, I think you copied the `"` signs, which I only used to split the characteristics from the sentence.

**Review by llogiq — COMMENTED** [review]
This is almost it... Just remove the `"`s from the characteristics.

File: tests/ui/wrong_self_convention.stderr (hunk 5 above)
Comment (mgacek8) [reply tests/ui/wrong_self_convention.stderr:42]: One thing I'm not sure about now, should to_* have backticks or not?
E.g. this way: 
error: methods with the following characteristics: (to_* and `self` type is not `Copy`) usually take self by reference
Or this: 
error: methods with the following characteristics: (`to_*` and `self` type is not `Copy`) usually take self by reference

Comment (llogiq) [reply tests/ui/wrong_self_convention.stderr:42]: I think it should be the latter, to drive home that this should match a method name. We could also backtick the `self` in "usually take `self` by reference".

**Comment by llogiq** [comment]
Thank you! @bors r+

**Pull request merged**